crates/core/src/surfnet/svm.rs

```diff
@@ -1207,6 +1202,38 @@ impl SurfnetSvm {
         self.slot_subscriptions
             .retain(|tx| tx.send(SlotInfo { slot, parent, root }).is_ok());
     }
+
+    pub fn subscribe_for_logs_updates(
+        &mut self,
+        commitment_level: &CommitmentLevel,
+        filter: &RpcTransactionLogsFilter,
+    ) -> Receiver<(Slot, RpcLogsResponse)> {
+        self.updated_at = Utc::now().timestamp_millis() as u64;
+        let (tx, rx) = unbounded();
+        self.logs_subscriptions
+            .push((commitment_level.clone(), filter.clone(), tx));
+        rx
+    }
+
+    pub fn notify_logs_subscribers(
+        &mut self,
+        signature: &Signature,
+        err: Option<TransactionError>,
+        logs: Vec<String>,
+        commitment_level: CommitmentLevel,
+    ) {
+        self.updated_at = Utc::now().timestamp_millis() as u64;
+        for (expected_level, filter, tx) in self.logs_subscriptions.iter() {
```

Comment:
Can we implement this filter? (specifically the `mentions` field)

Reply:
tracking this here: https://github.com/txtx/surfpool/issues/242
I need to spend some time understanding how that filtering is being done in the canonical implementation.